File: backend/src/agent_86/domain/models/session.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Any


@dataclass
class Session:
    id: str
    user_id: str
    title: str | None = None
    metadata: dict[str, Any] = field(default_factory=dict)
    created_at: datetime | None = None
    updated_at: datetime | None = None

File: frontend/src/types/session.ts
export interface Session {
  id: string;
  user_id: string;
  title: string | null;
  metadata: Record<string, unknown>;
  created_at: string | null;
  updated_at: string | null;
}

export interface CreateSessionRequest {
  title?: string | null;
  metadata?: Record<string, unknown>;
}

export interface UpdateSessionRequest {
  title?: string | null;
  metadata?: Record<string, unknown>;
}